test/users/vass/dont-eval-dot-type.future

```diff
@@ -1,12 +0,0 @@
-feature request: do not evaluate (at run time) the argument to .type
-
-This test argues that the argument to .type should not be evaluated
-at run time.
-
-This is a matter of definition. Currently, if the argument to .type is
-a function call, the compiler seems to be consistent in generating
-code to make that call at runtime.
-
-However, this is inconsistent with .type being a compile-time expression.
-Therefore I propose to define (and implement) .type as
-a purely compile-time construct, with no run-time effects.
```

Comment:
It may be worth eliminating this test altogether.
The tests you add in this PR definitely cover this.